client/lib/domains/dns/index.js

```diff
@@ -36,7 +36,7 @@ function validateField( { name, value, type, domainName } ) {
 		case 'name':
 			return isValidName( value, type, domainName );
 		case 'target':
-			return isValidDomain( value );
+			return isValidDomain( value ) || value === '.';
```

Comment:
Just curious: Why not incorporate the `value === '.'` check into the implementation of `isValidDomain`? It seems like that's what we're implying with this change anyway; that `.` is a valid domain. Seems like it would make the intent more clear.